### ISSUE TITLE
Agent status pending on openshift

### ISSUE THREAD

**Issue by tuananh** [litmuschaos/litmus#3325]
**What happened**:

`Self-Agent` is in pending status. 

`k get pod` returns only `server`, `frontend` and `mongo`

```sh
NAME                                    READY   STATUS    RESTARTS   AGE
litmuschaos-frontend-6859f7b84b-h8w5z   1/1     Running   0          137m
litmuschaos-mongo-0                     1/1     Running   0          137m
litmuschaos-server-64f6d9d784-njmcf     2/2     Running   0          137m
```

**What you expected to happen**:

i don't know where to check the logs. server logs indicate no error whatsoever.

**How to reproduce it (as minimally and precisely as possible)**:

install litmuschaos with helm

**Anything else we need to know?**:

openshift version `v1.20.0+87cc9a4`

`litmusctl create agent` failed and new agent shown as `pending`

```sh
litmusctl create agent
Project list:
1.  admin's project

Select a project [Range: 1-1]: 1

Installation Modes:
1. Cluster
2. Namespace

Select Mode [Default: cluster] [Range: 1-2]: 1

🏃 Running prerequisites check....
🔑 clusterrole ✅
🔑 clusterrolebinding ✅
🌟 Sufficient permissions. Installing the Agent...

Enter the details of the agent

Agent Name: my-agent

Agent Description: test

Do you want NodeSelector to be added in the agent deployments (Y/N) (Default: N):

Do you want Tolerations to be added in the agent deployments? (Y/N) (Default: N):

Platform List:
1. AWS
2. GKE
3. Openshift
4. Rancher
5. Others

Select a platform [Default: Openshift] [Range: 1-5]: 3

Enter the namespace (new or existing namespace) [Default: litmus]:
👍 Continuing with litmus namespace

Enter service account [Default: litmus]:

📌 Summary
Agent Name: my-agent
Agent Description: test
Platform Name: Openshift
Namespace:  litmus
Service Account:  litmus (new)

Installation Mode: cluster

🤷 Do you want to continue with the above details? [Y/N]: Y
👍 Continuing agent connection!!
Applying YAML:
http://<ingress-url>//api/file/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjbHVzdGVyX2lkIjoiNDU2OTJlNTUtZTg2NC00MjllLTkwZTgtMTkzZWVjMWVjN2ZkIn0.u4cw1N8Ny4pgwy0r-hahQtn-0bHOM6WY5ZxBNnd7O3Axxx.yaml

❌ Failed in applying connection yaml:
error: unable to read URL "http://<ingress-url>//api/file/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjbHVzdGVyX2lkIjoiNDU2OTJlNTUtZTg2NC00MjllLTkwZTgtMTkzZWVjMWVjN2ZkIn0.u4cw1N8Ny4pgwy0r-hahQtn-0bHOM6WY5ZxBNnd7O3Axxx.yaml", server reported 500 Internal Server Error, status code=500
```

logs from server at this point

```sh
2021/11/03 07:08:20 Server endpoint: http://<ingress-url>/backend/query
2021/11/03 07:08:20 error: open manifests/cluster: permission denied
```

**Comment by BK-STAR77** [comment]
Openshift 4.6.34

Hello we are encountering the exact same error

`2021/11/04 15:31:26 Server endpoint: http://x.x.x.x:30968/query`
`2021/11/04 15:31:26 error: open manifests/cluster: permission denied`

**Comment by gdsoumya** [comment]
@BK-STAR77 your issue seems to be similar to this one that can be found in our slack channel. It seems to be an issue with permission settings in openshift setup. Ref: https://kubernetes.slack.com/archives/CNXNB0ZTN/p1635992895275800?thread_ts=1635924872.263900&cid=CNXNB0ZTN

**Comment by tuananh** [comment]
The problem in my case is the security context constraint of openshift. You can verify it with exec into the pod and do a `ls`. it will fail.

To fix this, I need to add `anyuid` scc to the graphql server service account with the following command and restart the pod

```sh
oc adm policy add-scc-to-user anyuid -z <service-account-name> --as system:admin
```

**Comment by tuananh** [comment]
@BK-STAR77 here's how i fixed it https://github.com/litmuschaos/litmus/issues/3325#issuecomment-961510463

hope it works for you

**Comment by BK-STAR77** [comment]
> @BK-STAR77 here's how i fixed it [#3325 (comment)](https://github.com/litmuschaos/litmus/issues/3325#issuecomment-961510463)
> 
> hope it works for you

Perfect. Thank you
